src/Lykke.Service.KycReports.Services/Reports/KycReportingService.cs

```diff
@@ -497,30 +498,38 @@ private bool IsPending(KycStatusLogRecord row)
                 partnersDict[p.PublicId] = p.Name;
             }
 
+            Dictionary<string, ClientAccountInformationModel> clients = null;
+            if (items != null)
+            {
+                clients = (await _clientAccountService.GetClientsByIdsAsync(items.Select(_ => _.ClientId).Distinct().ToArray())).ToDictionary(_ => _.Id, _ => _);
+            }
+
             var auditLogEntities = (items)?
                             .Select(item =>
                             {
                                 var status = (KycStatus)item.CurrentStatus;
                                 var previousStatus = (KycStatus)item.PreviousStatus;
                                 
                                 string kycOfficer;
-                                if (item.Changer.StartsWith(_boChanger))
+                                if (item != null && item.Changer != null && item.Changer.StartsWith(_boChanger))
                                     kycOfficer = item.Changer.Substring(_boChanger.Length);
                                 else
                                     return null;
 
                                 var partnerName = _lykkeWalletPartnerName;
-                                var client = (_clientAccountService.GetByIdAsync(item.ClientId)).Result;
-                                if (client?.PartnerId != null)
-                                {
-                                    if (partnersDict.TryGetValue(client.PartnerId, out string name))
-                                    {
-                                        partnerName = name;
-                                    }
-                                    else
+                                ClientAccountInformationModel client;
+                                if (clients.TryGetValue(item.ClientId, out client)) {
+                                    if (client.PartnerId != null)
```

Comment:
Better use string.IsNullOrWhiteSpace nethod here - empty string also should be ignored.

Reply:
changed